src/metrics.jl

```diff
@@ -786,12 +788,6 @@ function _pairwise!(r::AbstractMatrix, dist::Euclidean, a::AbstractMatrix)
 end
 
 # Weighted Euclidean
-function colwise!(r::AbstractArray, dist::WeightedEuclidean, a::AbstractMatrix, b::AbstractMatrix)
-    sqrt!(colwise!(r, WeightedSqEuclidean(dist.weights), a, b))
-end
-function colwise!(r::AbstractArray, dist::WeightedEuclidean, a::AbstractVector, b::AbstractMatrix)
-    sqrt!(colwise!(r, WeightedSqEuclidean(dist.weights), a, b))
-end
```

Comment:
I understand these are redundant with the fallback methods? Does the fallback also avoids making a copy like `sqrt!`?

Reply:
The fallback simply computes the `sqrt` in the course of `evaluate` (in `eval_end`). The fallback might be even slightly faster, because it doesn't loop over the result array twice.